radio/src/pulses/afhds3.cpp

```diff
@@ -0,0 +1,661 @@
+            uint8_t id = telemetry[1];
+            if(id == 0xFE) id = 0xF7;  //use new id because format is different
+            if(length == 0 || telemetry + length > rxBuffer + rxBufferCount) break;
+            if(length == 4) { //one byte value fill missing byte
+              uint8_t data[] = { id, telemetry[2], telemetry[3], 0};
+              ::processFlySkySensor(data, 0xAA);
+            }
+            if(length == 5) {
+              if(id == 0xFA) telemetry[1] = 0xF8; //remap to afhds3 snr
+              ::processFlySkySensor(telemetry + 1, 0xAA);
+            }
+            else if(length == 6 && id == FRM302_STATUS) {
+              //convert to ibus
+              uint16_t t = (uint16_t)(((int16_t)telemetry[3] *10) + 400);
+              uint8_t dataTemp[] = { ++id, telemetry[2], (uint8_t)(t & 0xFF), (uint8_t)(t >> 8)};
+              ::processFlySkySensor(dataTemp, 0xAA);
+              uint8_t dataVoltage[] = { ++id, telemetry[2], telemetry[4], telemetry[5] };
+              ::processFlySkySensor(dataVoltage, 0xAA);
+            }
+            else if(length == 7) ::processFlySkySensor(telemetry + 1, 0xAC);
+            telemetry += length;
+          }
+        }
+      }
+      break;
+      case COMMAND::COMMAND_RESULT:
+      {
+          //AfhdsFrameData* respData = responseFrame->GetData();
+          //TRACE("COMMAND RESULT %02X result %d datalen %d", respData->CommandResult.command, respData->CommandResult.result, respData->CommandResult.respLen);
+      }
+      break;
+    }
+  }
+
+  if(responseFrame->frameType == FRAME_TYPE::REQUEST_GET_DATA || responseFrame->frameType == FRAME_TYPE::REQUEST_SET_EXPECT_DATA)
+  {
+    TRACE("Command %02X NOT IMPLEMENTED!", responseFrame->command);
+  }
+  else if (responseFrame->frameType == FRAME_TYPE::REQUEST_SET_EXPECT_ACK) {
+    //check if such request is not queued
+    if(!isEmpty()) {
+      Frame f = commandFifo[getIndex];
+      if(f.frameType == FRAME_TYPE::RESPONSE_ACK && f.frameNumber == responseFrame->frameNumber) {
+        TRACE("ACK for frame %02X already queued", responseFrame->frameNumber);
+        return;
+      }
+    }
+    /*
+    else {
+      Frame prev = commandFifo[prevIndex(getIndex)];
+      if(prev.frameType == FRAME_TYPE::RESPONSE_ACK && prev.frameNumber == responseFrame->frameNumber) {
+        TRACE("ACK for frame %02X already send", responseFrame->frameNumber);
+        return;
+      }
+    }
+    */
+    TRACE("AFHDS3 [QUEUE ACK] cmd %02X type %02X number %02X", responseFrame->command, responseFrame->frameType, responseFrame->frameNumber);
+    enqueueACK((enum COMMAND)responseFrame->command, responseFrame->frameNumber);
+  }
+  else if(responseFrame->frameType == FRAME_TYPE::RESPONSE_DATA || responseFrame->frameType == FRAME_TYPE::RESPONSE_ACK) {
+    if(operationState == State::AWAITING_RESPONSE) {
+      operationState = State::IDLE;
+    }
+  }
+}
+
+
+void PulsesData::trace(const char* message, uint8_t * payload, uint8_t payloadSize) {
```

Comment:
needed?

Reply:
DUMP does the same if I am not mistaken

Reply:
DUMP indeed is similar - function not used any-more - if needed DUMP combined with other TRACE can be used.